pkg/build/archive.go

```diff
@@ -27,7 +27,10 @@ import (
 // istioctl, and various tools.
 func Archive(manifest model.Manifest) error {
 	// First, build all variants of istioctl (linux, osx, windows). gen-charts is required for manifests compiled in to istioctl.
-	if err := util.RunMake(manifest, "istio", nil, "gen-charts", "istioctl-all", "istioctl.completion"); err != nil {
+	// Generate the binaries in a specific directory which is used later to copy from.
+	// Oould use RepoOutDIr, but then needs to remomve the "/release" from the end.
+	env := []string{"TARGET_OUT=" + manifest.RepoDir("istio") + "/out/linux_amd64"}
```

Comment:
sorry, haven't been following this closely - what is wrong with the default output location? isn't it always going to be the same as this?

Reply:
> /work/out/linux_amd64/release/istio-sidecar.deb which is where the file is created. But then the build tries to copy it from release/work/src/istio.io/istio/out/linux_amd64/release/istio-sidecar.deb

When we call `make deb` in istio/istio, we mount release/work/src/istio.io/istio as /work, So thse two paths should be identical.

Reply:
I think in CI we have BUILD_WITH_CONTAINER=0 at the top level, so we have no docker containers. Then if a user users docker container for builder, the docker env sets some variables which tamper with the individual repo's variables?

Reply:
The problem with archive binary building on my Mac is that the binaries are built in the /tmp/istio-release/../darwin_amd64/ directory (as the container builds into the directory to match the host OS). If one was to search through all the potential build OS directories (and maybe architectures) in the few places we copy binaries, this adds a bunch of code. This change causes the make to always put the binaries in the linux output directory. It should be safe as the makefile is specifying the go build architecture when creating the files and the build container is running linux.

Running in a container on a Linux host wouldn't require this change and the build and copy from directories would both point to .../linx...

Reply:
IN my container:
```
env | grep OUT
CONTAINER_TARGET_OUT=/work/out/darwin_amd64
TARGET_OUT_LINUX=/work/out/linux_amd64
CONTAINER_TARGET_OUT_LINUX=/work/out/linux_amd64
TARGET_OUT=/work/out/darwin_amd64
```

Reply:
I tried a different approach, and maybe this is better.

The issues with copying artifacts when being run in a build container had to do with various `make` artifacts be created in the container's output directory as set by various env vars. Instead of trying to tweak the individual `make` jobs to generate the artifacts in the manifest specified directory, I am `unsetting` the container env vars. This better emulates what happens in the prow jobs that don't use a container.